### PR TITLE
Deep Learning Regression

### PR DESCRIPTION
**Pull request by carlosmspk**
First project for deep learning: predict wine quality.
 - Created text file for non-code explanations (steps_log.txt)
 - Downloaded dataset from Kaggle.com (WineQT.csv)
 - Developed multiple "fully connected" neural network models using sklearn (main.py)
 - Stored model results (results.csv)
 - Developed script to plot results for multiple parameters (analysis.py)

### CODE REVIEW

File: DeepLearning/analysis.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+plot_relationships(res_df, "mean_test_score")
+
+#remove outlier
+res_df_filtered = res_df[res_df['mean_test_score'] > -3]
+
+plot_relationships(res_df_filtered, "mean_test_score")
+
+def plot_predictions(filename):
+    with open("DeepLearning/results/"+filename+".pickle", "rb") as f:
```
Comment (andre-exd) [inline DeepLearning/analysis.py]: Is it possible to format code using a formatter?

File: DeepLearning/analysis.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            distance_counts[distance] += 1
+        except KeyError:
+            distance_counts[distance] = 1
+    x = [key for key in distance_counts]
+    y = [value for value in distance_counts.values()]
+
+    plt.bar(x,y)
+    plt.show()
+
+plot_class_distances("data_lessfeatures_class")
```
Comment (andre-exd) [inline DeepLearning/analysis.py]: EOF

File: DeepLearning/main.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
+import pandas as pd
+
+do_gridsearch = False
+
+# store dataset as Data Frame (table format)
+df = pd.read_csv("DeepLearning/dataset/WineQT.csv", index_col="Id")
+print ("\n>>> First 5 rows of data:\n\n",df.head())
+
+
```
Comment (andre-exd) [inline DeepLearning/main.py]: Remove line

File: DeepLearning/main.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    "x_train" : x_train,
+    "x_test" : x_test,
+    "y_train" : y_train,
+    "y_test" : y_test,
+    "model": model
+}
+
+import pickle
+with open("DeepLearning/results/data_lessfeatures_class.pickle", "wb") as f:
+    pickle.dump(data, f)
```
Comment (andre-exd) [inline DeepLearning/main.py]: EOF

**Review by andre-exd — CHANGES_REQUESTED** [review]

**Review by andre-exd — CHANGES_REQUESTED** [review]
Please revert formatting existing files. You should only format the files you changed on this branch

File: DeepLearning/main.py (hunk 3 above)
Comment (andre-exd) [reply DeepLearning/main.py]: Seems you missed this one

File: DeepLearning/analysis.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
+import pandas as pd
+import pickle
+
+res_df = pd.read_csv("DeepLearning/results.csv", index_col="id")
+print(res_df.head())
+
+import matplotlib.pyplot as plt
+
```
Comment (andre-exd) [inline DeepLearning/analysis.py:8]: Seems like the formatter added 2 lines in a lot of places. Is that correct?
If not, please replace all 2 empty lines with a single one

**Review by andre-exd — CHANGES_REQUESTED** [review]

File: DeepLearning/analysis.py (hunk 5 above)
Comment (carlosmspk) [reply DeepLearning/analysis.py:8]: It's correct, it adds 2 lines before and after function definitions

**Review by andre-exd — APPROVED** [review]

**Pull request merged**